app/models/respondent.rb

```diff
@@ -0,0 +1,72 @@
+class Respondent < ApplicationRecord
+	include SpotlightSearch
+	include Paginator
+
+	belongs_to :user
+	belongs_to :response_round
+	belongs_to :organisation
+	has_many :consultation_response
+
+	scope :search_user_query, lambda { |query|
+		return nil if query.blank?
+		terms = query.downcase.split(/\s+/)
+		terms = terms.map { |e|
+			(e.gsub("*", "%").prepend("%") + "%").gsub(/%+/, "%")
+		}
+		num_or_conds = 3
+		includes(:user).references(:user).
+		where(
+			terms.map { |term|
+				"(LOWER(users.first_name) LIKE ? OR LOWER(users.last_name) LIKE ? OR LOWER(users.email) LIKE ? )"
+			}.join(" AND "),
+			*terms.map { |e| [e] * num_or_conds }.flatten,
+		)
+	}
+
+	def self.invite_respondent(consultation, organisation, params)
+		respondent_ids = params[:respondent][:ids].to_unsafe_h
+		if respondent_ids.present?
+			consultation_response_rounds = Consultation.includes(:response_rounds)
+			respondent_ids.each do |id, value|
+				respondent = Respondent.find(id.to_i)
+				response_round_ids = Respondent.where(user_id: respondent.user_id).map{|r| r.response_round_id }
+      	consultation_ids = consultation_response_rounds.where(response_rounds: { id: response_round_ids } ).map(&:id)
+				unless consultation_ids.include? consultation.id
+					Respondent.create(user_id: respondent.user_id, organisation_id: organisation.id, response_round_id: consultation.response_rounds.last.id)
+					if consultation.published?
+						url = self.respondent_invite_url(consultation, respondent.user)
+						InviteRespondentJob.perform_later(consultation, respondent.user, url)
+					end
+				end
+			end
+		end
+		emails = params[:respondent][:emails].split(",")
+    emails.each do |email|
```

Comment:
@BALAJI-BALA - you need to handle invites in separate model methods so that the code is easier to read.